code/controllers/stonedmc/subsystem/weeds.dm

```diff
@@ -0,0 +1,105 @@
+SUBSYSTEM_DEF(weeds)
+	name = "Weed"
+	priority = FIRE_PRIORITY_WEED
+	runlevels = RUNLEVEL_SETUP|RUNLEVEL_GAME
+	wait = 5 SECONDS
+
+	// This is a list of nodes on the map.
+	var/list/creating = list()
+	var/list/processing = list()
+	var/list/currentrun
+
+/datum/controller/subsystem/weeds/stat_entry()
+	return ..("Nodes: [length(processing)]")
+
+/datum/controller/subsystem/weeds/fire(resumed = FALSE)
+	if(!resumed)
+		currentrun = processing.Copy()
+		creating = list()
+
+	for(var/A in currentrun)
+		var/turf/T = A
```

Comment:
dont it like this just keep it in a simple processing list

Reply:
Do you mean process a single list?

If so, that wouldn't allow me to check for the health of a node. We add turfs on weed death for regrowth and they dont have a way to find their parent (without explicitly looking which id like to avoid).